course_access_groups/models.py

```diff
@@ -4,3 +4,103 @@
 """
 
 from __future__ import absolute_import, unicode_literals
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.text import slugify
+from django.utils.translation import ugettext as _
+from model_utils import models as utils_models
+from organizations.models import Organization
+from six import text_type
+
+
+class CourseAccessGroup(models.Model):
+    """
+    Group of students to determine which courses to show to them.
+
+    This model is organization-aware but can also work on single-site Open edX installations.
+    """
+
+    MAX_SLUGIFY_ATTEMPTS = 100  # Ensures the `get_unique_slug` won't run forever.
+
+    name = models.CharField(max_length=32)
+    description = models.CharField(
+        blank=True,
+        default='',
+        max_length=255,
+        help_text='An optional description about this group.'
+    )
+    slug = models.SlugField(unique=True)
+    organization = models.ForeignKey(Organization, null=True)
+
+    @classmethod
+    def generate_unique_slug(cls, org_name, group_name):
```

Comment:
@OmarIthawi I really don't like this and want to revisit this because it makes a set of calls to hunt for an open number. Yesterday I started to write a replacement that doesn't need the query loop or hardcoded max attempts, but had to leave for Django Boston. So I don't want to make this a blocker to get the rest of the code in this PR out.

You ok with me doing a PR to update this logic after this has been merged?

Reply:
Sure thing @johnbaldwin! I'd be happy to get any help on that. I would also suggest that you write your ideas in a GitHub issue so either you or I can get to it.